supervisor/api/snapshots.py

```diff
@@ -182,11 +182,15 @@ async def upload(self, request):
         """Upload a snapshot file."""
         with TemporaryDirectory(dir=str(self.sys_config.path_tmp)) as temp_dir:
             tar_file = Path(temp_dir, "snapshot.tar")
-
+            reader = await request.multipart()
+            contents = await reader.next()
             try:
                 with tar_file.open("wb") as snapshot:
-                    async for data in request.content.iter_any():
-                        snapshot.write(data)
+                    while True:
+                        chunk = await contents.read_chunk()
+                        if not chunk:
+                            break
+                        snapshot.write(chunk)
```

Comment:
This does I/O in the event loop. Probably also true for opening the file and opening and closing the temporary directory.

Not sure what's the best solution here. If we don't use context managers to handle the files we could open and close the files explicitly on the executor.